external-crates/move/crates/move-analyzer/src/code_action.rs

```diff
@@ -0,0 +1,428 @@
+// Copyright (c) The Move Contributors
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::{
+    completions::utils::{
+        all_mod_enums_to_import, all_mod_functions_to_import, all_mod_structs_to_import,
+        auto_import_text_edit, compute_cursor, import_insertion_info,
+    },
+    context::Context,
+    symbols::{
+        get_compiled_pkg, ChainInfo, CompiledPkgInfo, CursorContext, PrecomputedPkgInfo,
+        SymbolicatorRunner, Symbols,
+    },
+    utils::loc_start_to_lsp_position_opt,
+};
+
+use lsp_server::{Message, Request, Response};
+use lsp_types::{
+    CodeAction, CodeActionKind, CodeActionParams, Diagnostic, Position, Range, TextEdit,
+    WorkspaceEdit,
+};
+use move_symbol_pool::Symbol;
+use std::{
+    collections::{BTreeMap, BTreeSet, HashMap},
+    path::PathBuf,
+    sync::{Arc, Mutex},
+};
+use strum::IntoEnumIterator;
+use strum_macros::EnumIter;
+use url::Url;
+use vfs::VfsPath;
+
+use move_compiler::{
+    expansion::ast::ModuleIdent,
+    linters::LintLevel,
+    parser::ast::{LeadingNameAccess_, NameAccessChain_},
+    shared::{Identifier, Name},
+};
+use move_package::source_package::parsed_manifest::Dependencies;
+
+// The following reflects prefixes of error messages for
+// problems with a single-element access chain that are
+// subject to auto fixes. The constants representing them
+// cannot be lifted directly from the Move compiler,
+// as they are (at least most of them) dynamically generated
+// (in naming/translate.rs).
+#[derive(Debug, EnumIter)]
+enum SingleElementChainDiagPrefix {
+    UnboundType,
+    UnboundFunction,
+}
+
+impl SingleElementChainDiagPrefix {
+    fn as_str(&self) -> &'static str {
+        // we include `'` at the end of the prefix as some of the prefixes
+        // are shared with error messages that cannot be auto-fixed
+        match self {
+            SingleElementChainDiagPrefix::UnboundType => "Unbound type '",
```

Comment:
I think these are the only ones that are subject to auto-fixes. Others (e.g., `Unbound struct`, `Unbound struct or enum`, etc.) do not represent possible cross-module dependency (as, I think, they are used in context of internal module operations such as pack or unpack). I may be wrong here, though, and I am happy to add more cases here if this is indeed true

Reply:
I could see wanting to do it for two-element chains, but it would require more machinery.